Manak_Leah_ndvi_2022.py

```diff
@@ -0,0 +1,670 @@
+#     * The code to create a plot of mean NDVI across a year for  2 NEON Field Sites:
+#         * NDVI on the x axis and formatted dates on the y for both NEON sites on one figure/axis object
+#     * The **data should be cleaned to remove the influence of clouds**. See the [earthdatascience website for an example of what your plot might look like with and without removal of clouds](https://www.earthdatascience.org/courses/earth-analytics-python/create-efficient-data-workflows/).
+# * BONUS: Create one output `.csv` file that has 3 columns - NDVI, Date and Site Name - with values for SJER and HARV.
+# 
+# Your notebook should:
+# * Have *at least* 2 well documented and well named functions with docstrings.
+# * Include a Markdown cell at the top of the notebook that outlines the overall workflow using pseudocode (i.e. plain language, not code)
+# * Include additional Markdown cells throughout the notebook to describe: 
+#     * the data that you used - and where it is from
+#     * how data are being processing
+#     * how the code is optimized to run fast and be more concise
+
+# # Replace this cell with your pseudocode  for this workflow
+# 
+# If you happen to be a diagram person a diagram is ok too
+# 
+# 
+
+# # Pseudocode for assignment workflow
+# 
+# - Import libraries that will be used in this notebook
+# - Make sure unused libraries are not in the list
+# - Get the home directory using "et.data.get_data('ndvi-automation')"
+# - Create data paths to our data from 'ndvi-automation' downloaded from earthpy
+# - Create cloud mask and get cloud pixel values from earthpy
+# - Create two functions: 
+#     1. extract the sitename and datetime from pathnames
+#     2. open, crop, and identify ranges of a landsat band
+# - Clip extent of bands to match study sites boundaries
+# - Clip extent of qa layers to match study sites boundaries
+# - Calculate NDVI 
+# - Add a cloud mask to the NDVI values
+# - Calculate Mean NDVI, get site names and the dates, and create a list of lists
+# - Make a pandas dataframe including the site names, mean NDVIs, and dates 
+# - Make sure the dataframe index is in datetime and that the date column is the index
+# - Make sure the NA data is not included
+# - Plot the data with date on the x-axis and mean NDVI values on the y axis
+# - Download the data to a CSV file
+
+# In[1]:
+
+
+# Autograding imports - do not modify this cell
+import matplotcheck.autograde as ag
+import matplotcheck.notebook as nb
+import matplotcheck.timeseries as ts
+from datetime import datetime
+
+
+# In[2]:
+
+
+# Import needed packages in PEP 8 order
+# and no unused imports listed (10 points total)
+import os
+from glob import glob
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import rioxarray as rxr
+import xarray as xr
+import geopandas as gpd
+import earthpy as et
+import earthpy.mask as em
+from datetime import datetime
+import numpy as np
+from matplotlib.dates import DateFormatter
```

Comment:
@LManak great work only importing packages that are actually used in the document